sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExec.scala

```diff
@@ -1782,11 +1782,22 @@ class GpuCachedDoublePassWindowIterator(
   }
 
   // Compute the window operation and cache/update caches as needed.
+  // This method takes ownership of cb
   def firstPassComputeAndCache(cb: ColumnarBatch): Unit = {
     val fixers = fixerIndexMap
     val numRows = cb.numRows()
-    withResource(computeBasicWindow(cb)) { basic =>
-      withResource(GpuProjectExec.project(cb, boundPartitionSpec)) { parts =>
+
+    val sp = SpillableColumnarBatch(cb, SpillPriorities.ACTIVE_BATCHING_PRIORITY)
+    val (basic, parts) = withRetryNoSplit(sp) { _ =>
+      withResource(sp.getColumnarBatch()) { x =>
+        closeOnExcept(computeBasicWindow(x)) { basic =>
+          (basic, GpuProjectExec.project(x, boundPartitionSpec))
+        }
+      }
+    }
```

Comment:
nit
```suggestion
    val (basic, parts) = withRetryNoSplit(sp) { _ =>
      withResource(sp.getColumnarBatch()) { cb =>
        closeOnExcept(computeBasicWindow(cb)) { basic =>
          (basic, GpuProjectExec.project(cb, boundPartitionSpec))
        }
      }
    }
```

Reply:
I pushed a commit to rename this one to `batch` to avoid ambiguity with the existing `cb` variable.